unity_buy_android/unity_buy_plugin/src/main/java/com/shopify/unity/buy/AndroidPayCheckoutSession.java

```diff
@@ -0,0 +1,102 @@
+package com.shopify.unity.buy;
+
+import android.util.Log;
+
+import com.shopify.buy3.pay.PayCart;
+import com.unity3d.player.UnityPlayer;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.math.BigDecimal;
+
+public class AndroidPayCheckoutSession {
+
+    public boolean checkoutWithAndroidPay(
+            String unityDelegateObjectName,
+            String merchantName,
+            String summaryItemsString,
+            String currencyCode,
+            String countryCode,
+            boolean requiresShipping) {
+
+        SummaryItems items = null;
+        try {
+            items = SummaryItems.fromJsonString(summaryItemsString);
+        } catch (JSONException e) {
+            Log.e("ShopifyBuyPlugin", "Failed to parse summary items from Unity");
+            e.printStackTrace();
+            return false;
+        } catch (IllegalArgumentException e) {
+            Log.e("ShopifyBuyPlugin", e.getMessage());
+            e.printStackTrace();
+            return false;
+        }
+
+        PayCart cart = PayCart.builder()
+                .merchantName(merchantName)
+                .currencyCode(currencyCode)
+                .shippingAddressRequired(requiresShipping)
+                .subtotal(items.subtotal)
+                .shippingPrice(items.shippingPrice)
+                .taxPrice(items.taxPrice)
+                .totalPrice(items.totalPrice)
+                .build();
+
+        addAndroidPayFragment(unityDelegateObjectName, cart);
+        return true;
+    }
+
+    private void addAndroidPayFragment(String unityDelegateObjectName, PayCart cart) {
+        UnityAndroidPayFragment payFragment =
+                UnityAndroidPayFragment.newInstance(unityDelegateObjectName, cart);
+        UnityPlayer.currentActivity.getFragmentManager()
+                .beginTransaction()
+                .add(payFragment, "payFragment")
+                .commit();
+    }
+
+    // Wrapper around SummaryItems from Unity that performs validation.
+    private static class SummaryItems {
+        private BigDecimal subtotal;
```

Comment:
What do you mean?
`private static class SummaryItems`

My comment was about redundant private access modifiers for fields `private BigDecimal subtotal`